### DISCUSSION TITLE
Boot both x32 and x64 clients

### DISCUSSION THREAD

**Discussion by ArtAskGit** [ltsp/ltsp#663] [Q&A] [answered]
Hello ,

          I recently started using the LTSP for diskless booting.
The LTSP clients and server both have x64 architecture.
The server is using Ubuntu x64.
The LTSP image was generated on the same server.
This network of clients works great!
          
          But recently there has been a need to add diskless clients with x32 architecture to this network. 
The  x32 clients  cannot boot from the x64 server because  the image uses x64 Ubuntu.
 I can try to fix this by  generating an image on  a x32 Debian OS and possibly add it to my x64 server.
 Is this enough to boot both x32 and x64 clients?

Could you suggest the correct way to configure x64 server to boot both x32 and x64 clients at the same time?

Thanks in advance!
Art.

**Comment by alkisg** [comment]
Hello Art,

Sure, you can create a VirtualBox VM with either Ubuntu 18.04 32bit or Debian Bullseye 32bit and symlink it e.g. like:


```
ln -rs ~/VirtualBox\ VMs/debian/debian-flat.vmdk /srv/ltsp/x86_32.img
```

If you name the symlink "x86_32", it will be automatically selected for 32bit clients.
Read the documentation on how to create the VM: https://ltsp.org/man/ltsp-image/#vm-images

**Reply by ArtAskGit** [reply]
As I understand after symlink I need to export this image to the clients:
`ltsp image x86_32.img`

and probably Generate ltsp.img by the following command:
`ltsp initrd`

Is it right?

**Reply by ArtAskGit** [reply]
Hello Alkis,

> Hello Art,
> 
> Sure, you can create a VirtualBox VM with either Ubuntu 18.04 32bit or Debian Bullseye 32bit and symlink it e.g. like:
> 
> ln -rs ~/VirtualBox\ VMs/debian/debian-flat.vmdk /srv/ltsp/x86_32.img
> If you name the symlink "x86_32", it will be automatically selected for 32bit clients.
> Read the documentation on how to create the VM: https://ltsp.org/man/ltsp-image/#vm-images
>


Can I do this without virtual machines?
As I undersatnd it, LTSP needs two images: x86_32.img and ltsp.img (see attached image for 32-bit OS).


For example, I can put the x86_32.img and x86_64.img iimages in srv/ltsp/images/ folder.
But ltsp.img (in /srv/tftp/ltsp/ folder) doesn't have OS type in the name (32-bits or 64-bits). 
How to name ltsp.img so that LTSP can understand for which OS (64-bit or 32-bit) this file is intended?

Thank you in advance
![image](https://user-images.githubusercontent.com/99243964/155135805-328c7b17-6c37-451e-9808-4ea42fbe6bb0.png)
.

**Comment by alkisg** [comment]
These should be enough:

```shell
sudo ltsp image x86_32
sudo ltsp ipxe
```

**Comment by alkisg** [comment, marked as answer]
@ArtAskGit try to reply at the bottom so that we can follow the conversation. These "threads" that github decided to mimic are awful.

If you have a VM image, you do:

```shell
ln -rs ~/VirtualBox\ VMs/debian/debian-flat.vmdk /srv/ltsp/x86_32.img
ltsp image x86_32
ltsp ipxe
```

If you have a squashfs image that you prepared elsewhere, you do:

```
mv source.squashfs /srv/ltsp/images/x86_32.img
ltsp kernel x86_32
ltsp ipxe
```

What do you have, a VM image that goes in /srv/ltsp, or a squashfs image that goes in /srv/ltsp/images?

Note that the /srv/tftp/ltsp/ltsp.img image is common in all cases, it's not different per architecture. It only contains the shell LTSP code, the server user accounts etc, it doesn't contain anything from the VM or the image.

**Reply by ArtAskGit** [reply]
I have an image in srv/ltsp/images/ folder.
I think of adding the x86_32.img image for 32-bit clients and x86_64.img for 64-bit clients to the srv/ltsp/images/ folder.
